pkg/apis/servicecatalog/types.go

```diff
@@ -81,6 +81,17 @@ type ClusterServiceBrokerSpec struct {
 	// RelistRequests is a strictly increasing, non-negative integer counter that
 	// can be manually incremented by a user to manually trigger a relist.
 	RelistRequests int64
+
+	// CatalogRestrictions allows adding restrictions onto Class/Plans on relist.
+	CatalogRestrictions *ServiceClassCatalogRestrictions
+}
+
+type ClusterServicePlanRequirements string
+type ClusterServiceClassRequirements string
+
+type ServiceClassCatalogRestrictions struct {
+	ServicePlan  ClusterServicePlanRequirements
```

Comment:
It's not clear to me how I would say:

- From this broker, create resources only for service X
- From this broker, create resources for all services but Y

Reply:
I have added some more comments to help you understand.